### DISCUSSION TITLE
[server] 1주차 -  7/16일까지 할일

### DISCUSSION THREAD

**Discussion by 2020147542** [BookMoA/server#1] [General]
다음주까지 해야할 일을 정해봅시다 ✨

### 7/9(화) 역할분담
- notion: 이다솔
- 깃헙: 강슬미
- 카톡: 이다윤
- ERD CLOUD: 김지민

**Comment by 2020147542** [comment]
### 공통으로 해야할 일

1. 기능정리 후 api 문서 작성하기 
2. ERD 그리기
3. 깃헙 컨벤션 정하기(pr/commit + 템플릿까지 적용)
4. 초기세팅 + CI 포함

**Comment by 2020147542** [comment]
### 3. 깃헙 컨벤션 정하기

찾아보니까 깃헙 컨벤션 강제화 하는 기능들이 있는데
아래 링크 한번 참고해서 저희도 이런거 쓰면 좋을 것 같아요!
[컨벤션 포맷 도구](https://tech-marineplay.tistory.com/10)

**Reply by 2020147542** [reply]
### 해야할 일&버그 발견시 이슈 등록하기 
- 이슈 제목: `[feature/reafactor/fix...등등] 할일` 작성
- 이슈 내용: 채크박스 사용해서 todo 리스트 작성 -> 진행 완료한 일은 체크하기
- assignee: 담당자 할당하기
- tag: 관련된 작업 등록(깃모지 사용)
- branch: 태그/이슈번호 ex) `feature/13`
- 참고사항 및 기타: 유의할 점,,, 알아서 적기

### commit 메세지 컨벤션 
- commit: `[feature/reafactor/fix...등등] 한 일`

### PR 제목 컨벤션 
- 제목: comiit 메세지와 동일
- description: 변경 사항 작성
- 참고사항 및 기타: 유의할 점,,, 알아서 적기
- merge시: Squash & Merge 방식으로 진행!
- reviewer : 관련 기능 담당자 지정 또는 임의로 지정 후 카톡에 알림

**Reply by dasol729** [reply]
- [x]  Commit Message Convention에 따른 label 추가하기

#### 📃 Commit Message Convention
   커밋명 | 내용
   -- | --
   ✨ feature | 파일, 폴더, 새로운 기능 추가
   🐛 fix | 버그 수정
   🎨 style | 코드 포맷 변경, 세미 콜론 누락, 코드 수정이 없는 경우
   ♻️ refactor | 프로덕션 코드 리팩토링
   💡 comment | 필요한 주석 추가 및 변경
   📝 docs | README 등의 문서 수정
   🔧 chore | 환경설정, 빌드 업무, 패키지 매니저 설정 등(프로덕션 코드 변경 X)
   ✏️ rename | 파일 혹은 폴더명을 수정하거나 옮기는 작업만 수행한 경우
   🔥 remove | 파일 등을 삭제하는 작업만 수행한 경우
   🚑️ !hotfix | 급하게 치명적인 버그를 고쳐야하는 경우

**Comment by 2020147542** [comment]
### 2. ERD 그리기

각자 맡은 기능들 기준으로 [ERD CLOUD](https://www.erdcloud.com/d/yzgQueER2cwtpRmFt)에 추가하면서 
7/16일 회의 때 확인 및 의논하기로 결정

**Reply by jinnieusLab** [reply]
https://www.erdcloud.com/d/yzgQueER2cwtpRmFt
erd cloud 링크입니다!

**Comment by 2020147542** [comment]
### 4. 초기세팅 + CI 포함 dev branch에 올리기

- [x] swagger [merged #6](https://github.com/BookMoA/server/pull/6)
- [x] issue/pr template CI [merged #2](https://github.com/BookMoA/server/pull/2), [merged #5 ](https://github.com/BookMoA/server/pull/5), [merged #6](https://github.com/BookMoA/server/pull/6)
- [x] .gitignore 파일 추가 [merged #6](https://github.com/BookMoA/server/pull/6)
- [x] build 확인 CI 
- [x] commit/pr convention CI [merged #12](https://github.com/BookMoA/server/pull/12)
- [x] .git hook setting - local에서의 commit (포맷팅), Spotless (코드 포맷팅) [merged #12](https://github.com/BookMoA/server/pull/12)

**Comment by 2020147542** [comment]
### 1. 기능분담
IA: 홈+기록, 서재, 독서모임, 설정+로그인
<img width="800" alt="image" src="https://github.com/BookMoA/server/assets/80676180/f96c7e8a-b65b-4059-88ae-5e9617d4ce96">

이다솔 : 독서모임
강슬미 : 설정+로그인
이다윤 : 서재
김지민 : 홈+기록
